### ISSUE TITLE
Remote query types for `belongsTo` relationship 

### ISSUE THREAD

**Issue by noubase** [medusajs/medusa#9922]
## Bug report

Improper type for a 'belongsTo' relationship

### Describe the bug

Type should be `Maybe<T>` instead of `Array<Maybe<T>>`

### System information

Medusa version (including plugins): v 2.0.1
Node.js version:
Database:
Operating system:
Browser (if relevant):

### Steps to reproduce the behavior

Create an example model

```
const PackingCollectionVariant = model.define(
	'packing_collection_variant',
	{
		id: model.id({ prefix: 'pckcolvar_' }).primaryKey(),
		variant: model.text().index(),
	        ...
		collection: model.belongsTo(() => PackingCollection, {
			mappedBy: "variants"
		}),
	})
.indexes([
	{
		on: ['collection_id', 'sku'],
		unique: true,
	}
])

export default PackingCollectionVariant
```

Generated type will be 

```
export type PackingCollectionVariant = {
  __typename?: 'PackingCollectionVariant';
  id: Scalars['ID']['output'];
  variant: Scalars['String']['output'];
  ...
  collection: Array<Maybe<PackingCollection>>;
  ...
};
```

### Expected behavior

It should be

```
export type PackingCollectionVariant = {
  __typename?: 'PackingCollectionVariant';
  id: Scalars['ID']['output'];
  variant: Scalars['String']['output'];
  ...
  collection: Maybe<PackingCollection>;
  ...
};
```

**Comment by adrien2p** [comment]
Hi @noubase , I have tried to reproduce you example (see screenshot 1) and I cant get it to fail, would you be able to share a little repo that reproduces the issue so that I can play with it please?

<img width="1681" alt="Screenshot 2024-11-05 at 09 56 51" src="https://github.com/user-attachments/assets/cfd6ed44-2a0e-4664-b5b6-d6811dd6225e">

**Comment by noubase** [comment]
[models.zip](https://github.com/user-attachments/files/17630027/models.zip)
Sorry, I don’t have time right now. Here are my models.

**Comment by noubase** [comment]
One more question — how do I rebuild those types from the command line? Should I use `npx medusa build`?

**Comment by adrien2p** [comment]
Thank you for sharing it, I ll have a look at it. In regards to the types generation, for now, they are re-generated every time you `start` or the `develop` command re-start.

**Comment by noubase** [comment]
I also found an issue with `OrderLineItem`. There is no `product` field, but I can resolve it with `query.graph`

<img width="828" alt="image" src="https://github.com/user-attachments/assets/e2ff1f73-6a44-44be-b25a-7cf77d3ff9b6">

**Comment by adrien2p** [comment]
this [pr](https://github.com/medusajs/medusa/pull/9932) should fix your issue with the blongs to . Let me know

**Comment by adrien2p** [comment]
this [pr](https://github.com/medusajs/medusa/pull/9932) should fix your issue with the belongs to. Let me know 👍

**Comment by adrien2p** [comment]
In regards to that `product` issue, it is link to the type generation, we will fix it (I am currently discussing and investigating with @carlos-r-l-rodrigues about it and we found the TODO we need to resolve to fix that).  Can I ask you to fill up a new issue specifically about this and then we can probably close this one once the other pr is merged?

**Comment by noubase** [comment]
> Thank you for sharing it, I ll have a look at it. In regards to the types generation, for now, they are re-generated every time you `start` or the `develop` command re-start.

When I use remote-query-entry-points for types like this:

`import {ProductStatus, ProductVariant} from "../../../.medusa/types/remote-query-entry-points";`

and then drop the .medusa folder, I can’t start npx medusa develop again due to the following error:

```
error:   Error starting server
Error: Cannot find module '../../../.medusa/types/remote-query-entry-points'
```

So I can't use them?

**Comment by noubase** [comment]
I thought those types should be built beforehand, right?

**Comment by adrien2p** [comment]
We use to have a separate cli command to generate them independently. I personally feel that it would indeed be nice to have it to not fall into those scenarios. Now just for info, you don't have to remove the `.medusa` folder manually, what is the reason for doing this? 

cc @thetutlage wdyt about a separate command to generate the types?

PS: I believe this discussion should be moved to a separate issue though

**Comment by noubase** [comment]
The `.medusa` folder can be deleted accidentally, and the main issue is that there’s currently no way to `start` the project if that happens. It would be great to have a separate CLI command to generate types. Additionally, these types should be generated beforehand automatically when launching with the `start` or `develop` command. I’m lucky that IntelliJ IDEA has its own local history feature, allowing me to revert the folder, but most IDEs don’t have this feature, and the `.medusa` folder is usually added to `.gitignore` by default.

**Comment by noubase** [comment]
> In regards to that `product` issue, it is link to the type generation, we will fix it (I am currently discussing and investigating with @carlos-r-l-rodrigues about it and we found the TODO we need to resolve to fix that). Can I ask you to fill up a new issue specifically about this and then we can probably close this one once the other pr is merged?

https://github.com/medusajs/medusa/issues/9952

**Comment by jschill** [comment]
> We use to have a separate cli command to generate them independently. I personally feel that it would indeed be nice to have it to not fall into those scenarios. Now just for info, you don't have to remove the `.medusa` folder manually, what is the reason for doing this?
> 
> cc [@thetutlage](https://github.com/thetutlage) wdyt about a separate command to generate the types?
> 
> PS: I believe this discussion should be moved to a separate issue though

@adrien2p Was this discussed somewhere? I need to have those files generated, for the `npm run build` to work and the only way to generate those files are via `medusa start --types=true` which is after the build step?
